src/config.py

```diff
@@ -0,0 +1,296 @@
+from dune_client.types import Address
+from gnosis.eth.ethereum_network import EthereumNetwork
+from web3 import Web3
+
+load_dotenv()
+
+
+class Network(Enum):
+    """Network class for networks supported by the accounting."""
+
+    MAINNET = "mainnet"
+    GNOSIS = "gnosis"
+    ARBITRUM_ONE = "arbitrum"
+    BASE = "base"
+
+
+@dataclass(frozen=True)
+class RewardConfig:
+    """Configuration for reward mechanism."""
+
+    reward_token_address: Address
+    cow_bonding_pool: Address
+    batch_reward_cap_upper: int
+    batch_reward_cap_lower: int
+    quote_reward_cow: int
+    quote_reward_cap_native: int
+    service_fee_factor: Fraction
+
+    @staticmethod
+    def from_network(network: Network) -> RewardConfig:
+        """Initialize reward config for a given network."""
+        match network:
+            case Network.MAINNET:
+                return RewardConfig(
+                    reward_token_address=Address(
+                        "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+                    ),
+                    batch_reward_cap_upper=12 * 10**15,
+                    batch_reward_cap_lower=10 * 10**15,
+                    quote_reward_cow=6 * 10**18,
+                    quote_reward_cap_native=6 * 10**14,
+                    service_fee_factor=Fraction(15, 100),
+                    cow_bonding_pool=Address(
+                        "0x5d4020b9261f01b6f8a45db929704b0ad6f5e9e6"
+                    ),
+                )
+            case _:
+                raise ValueError(f"No reward config set up for network {network}.")
+
+
+@dataclass(frozen=True)
+class ProtocolFeeConfig:
+    """Configuration for protocol and partner fees."""
+
+    protocol_fee_safe: Address
+    partner_fee_cut: float
+    partner_fee_reduced_cut: float
+    reduced_cut_address: str
+
+    @staticmethod
+    def from_network(network: Network) -> ProtocolFeeConfig:
+        """Initialize protocol fee config for a given network."""
+        match network:
+            case Network.MAINNET:
+                return ProtocolFeeConfig(
+                    protocol_fee_safe=Address(
+                        "0xB64963f95215FDe6510657e719bd832BB8bb941B"
+                    ),
+                    partner_fee_cut=0.15,
+                    partner_fee_reduced_cut=0.10,
+                    reduced_cut_address="0x63695Eee2c3141BDE314C5a6f89B98E62808d716",
+                )
+            case _:
+                raise ValueError(
+                    f"No protocol fee config set up for network {network}."
+                )
+
+
+@dataclass(frozen=True)
+class BufferAccountingConfig:
+    """Configuration for buffer accounting."""
+
+    include_slippage: bool
+
+    @staticmethod
+    def from_network(network: Network) -> BufferAccountingConfig:
+        """Initialize buffer accounting config for a given network."""
+        match network:
+            case Network.MAINNET:
+                return BufferAccountingConfig(include_slippage=True)
```

Comment:
Does this bypass the flag that is passed as runtime argument?

Reply:
Currently, the code uses `ignore_slippage_flag or (not config.buffer_accounting_config.include_slippage)`. So if one of them is set to ignore slippage, the slippage part is skipped.

I have a PR almost ready locally to pass the config explicitly everywhere. Then the config is created after parsing of command line arguments and the config object would be set based on the command line flag.